pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -347,6 +351,14 @@ stdenv.mkDerivation ({
 
     isHaskellLibrary = hasActiveLibrary;
 
+    # TODO: ask why the split outputs are configurable at all?
+    # TODO: include tests for split if possible
+    # Given the haskell package, returns
+    # the directory containing the haddock documentation.
+    # `null' if no haddock documentation was built.
+    # TODO: fetch the self from the fixpoint instead
+    haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
```

Comment:
I don't want this attribute to be part of every Haskell package's "API". If you want to abstract the computation of this path in some re-usable fashion, then please add an appropriate function to the `lib.nix` module and re-use it in all places that depend on that path.